plugins/source/vault/client/spec.go

```diff
@@ -0,0 +1,30 @@
+package client
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+)
+
+const defaultConcurrency = 10000
+
+type Spec struct {
+	VaultAddress string `json:"vault_address,omitempty"`
+	Concurrency  int    `json:"concurrency"`
+}
+
+func (s *Spec) SetDefaults() {
+	if s.Concurrency == 0 {
+		s.Concurrency = defaultConcurrency
+	}
+}
+
+func (s *Spec) Validate() error {
+	if s.VaultAddress == "" {
+		return errors.New("no vault address provided")
+	}
+	if _, err := url.ParseRequestURI(s.VaultAddress); err != nil {
+		return fmt.Errorf("invalid vault address provided %q", s.VaultAddress)
```

Comment:
```suggestion
		return fmt.Errorf("invalid vault address provided %q: %w", s.VaultAddress, err)
```

The error message might be useful to users for debugging what's wrong